### ISSUE TITLE
fix: pcanormalizer.transform centres each batch on its own mean instead of the training mean

### ISSUE THREAD

**Issue by g8rdier** [g8rdier/hqnn-forge#64]
## Context

`PCANormalizer.transform` standardises each batch with that batch's own mean, not with a
statistic learned during `fit`:

```python
# hqnn_forge/preprocessing/pca_normalizer.py:197
standardised = (projections - projections.mean(axis=0)) / self.std_
```

`fit` already centres the data on the training mean (`self.mean_`) before projecting, so the
projections of the training data have zero mean by construction. Subtracting
`projections.mean(axis=0)` is therefore a no-op when `transform` is called on the training
set, and wrong for every other input:

- **A single row always comes out as all zeros.** Its batch mean equals the row itself, so
  every one-sample prediction feeds zero angles to the circuit, whatever the input.
- **The output depends on which other rows are in the batch.** The same sample gets different
  encoded angles in a batch of 2, a batch of 50, or the full test set.
- **Test data is re-centred on itself, not on the training distribution.** Any systematic
  shift between train and test is silently removed before the model sees it.

The docstring for `std_` says it is "computed on training projections", which matches the
intent of reusing training statistics. The mean subtraction does not. This does not raise; it
just produces wrong angles.

## Steps to Reproduce

```python
import numpy as np
from hqnn_forge.preprocessing import PCANormalizer

rng = np.random.default_rng(0)
X = rng.standard_normal((200, 10)) + 3.0
p = PCANormalizer(n_components=4).fit(X)

print(p.transform(X[:1]))      # tensor([[0., 0., 0., 0.]])
print(p.transform(X[:2])[0])   # tensor([-0.2125, -0.1689,  1.3845, -2.8002])
print(p.transform(X[:50])[0])  # tensor([ 0.5124,  1.2912,  2.7099, -2.3628])
print(p.transform(X)[0])       # tensor([ 0.9384,  0.7431,  2.4998, -2.3920])
```

Row 0 gets four different encodings depending on batch size, and zeros when on its own.

## Affected Code

- `hqnn_forge/preprocessing/pca_normalizer.py:197`, the mean subtraction in `transform`.
- `examples/quick_start.py:116`, `pca.transform(X_test_np)`: the test set is centred on its
  own mean.
- `tests/test_integration.py:78`, same pattern in `test_predict_proba_after_training`.
- `tests/test_preprocessing.py:46-55`: these tests only transform the training data, where
  the batch mean equals the training mean, so they cannot catch this.

## Expected Outcome

- `transform` uses only statistics learned in `fit`: centre on `self.mean_`, project, divide
  by `self.std_`, with no per-batch mean. `fit_transform` output on the training set should be
  unchanged, up to floating-point error.
- Tests that assert the numbers, not just shapes:
  - transforming a single row gives the same result as that row inside a larger batch;
  - transforming a held-out batch does not depend on how it is split;
  - transforming a single non-mean row is not all zeros.

## Type

Bug: silently corrupts encoded features for any input other than the exact training set.
Introduced in 7e95480; found during review of the fix for #57.
